tests/test_merkle_tree.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+
+from merkle_zeppelin import MerkleTree
+
+
+@pytest.mark.parametrize(
+    "leafs, encoded_leafs",
+    [
+        (
+            [(1,), (2,), (3,), (4,), (5,)],
+            [
+                "c167b0e3c82238f4f2d1a50a8b3a44f96311d77b148c30dc0ef863e1a060dcb6",
+                "b5d9d894133a730aa651ef62d26b0ffa846233c74177a591a4a896adfda97d22",
+                "2584db4a68aa8b172f70bc04e2e74541617c003374de6eb4b295e823e5beab01",
+                "1ab0c6948a275349ae45a06aad66a8bd65ac18074615d53676c09b67809099e0",
+                "16db2e4b9f8dc120de98f8491964203ba76de27b27b29c2d25f85a325cd37477",
+            ],
+        ),
+        ([(1,)], ["b5d9d894133a730aa651ef62d26b0ffa846233c74177a591a4a896adfda97d22"]),
+    ],
+)
+def test_leafs_hashes(leafs: list[tuple[int]], encoded_leafs: list[bytes]) -> None:
+    # when
+    merkle = MerkleTree(leafs, ["int256"])
+
+    # then
+    assert [leaf.hex() for leaf in merkle.leafs[: len(encoded_leafs)]] == encoded_leafs
+
+
+@pytest.mark.parametrize(
+    "leafs_number, duplication_number", [(3, 1), (5, 3), (1, 0), (9, 7)]
+)
+def check_last_leaf_duplication(leafs_number: int, duplication_number: int) -> None:
+    # given
+    leafs = [(1,) * (leafs_number - 1)] + [(2,)]
+    encoded_duplicat = (
```

Comment:
Tiny typo maybe - didn't you mean "duplicate"?

Reply:
yeah, thanks for highlighting that!